### PR TITLE
Added Loading Screen with Success/Error Feedback for Registration and Login

### LINKED ISSUE #21: Implement Loading Screen during User Registration
## DESCRIPTION

This issue involves implementing a loading screen to provide visual feedback to users while their registration is being processed. The loading screen will be displayed after the user submits their registration details and before the registration process is completed.

```[tasklist]
### Tasks
- [ ] Design the Loading Screen.
- [ ] Implement Loading State.
- [ ] Integrate the loading screen with the user registration process.
```

### PR DESCRIPTION
**Pull request by DerekAyala**
# Description

This issue involves improving the user experience during the registration and login process by adding a loading screen. The loading screen should provide feedback on the success or failure of the action, displaying different icons and messages accordingly.

# changes made:
- Created a new component called "LoadingModal" to serve as a loading screen during the registration and login process.
- Implemented the necessary logic to display the loading screen when a registration or login request is in progress.
- Designed the loading screen to show a spinner or loading animation while the action is being processed.
- Added functionality to display a success or error icon, along with an appropriate message, upon completion of the registration or login process.
- Integrated the LoadingModal component into the registration and login flows to provide a seamless and responsive user experience with feedback on the outcome of the action.
# Testing
## Email sign in
email sign in
Screenshot_1688695149
Screenshot_1688695150

## Email Sign up error
Screenshot_1688622227
Screenshot_1688622238
Screenshot_1688622232
## Email Sign up
Screenshot_1688622317

### CODE REVIEW

**Review by Tole15 — APPROVED** [review]

**Comment by Tole15** [comment]
Hey, buen trabajo 👍

**Review by BrandonVG — APPROVED** [review]

File: AppVuelos/src/screens/register.tsx [hunk 1]
```diff
@@ -39,25 +44,49 @@ function Register(): JSX.Element {
...
     createUserWithEmailAndPassword(auth, email, password)
-      .then((userCredential) => {
-      console.log(userCredential)
-      navigation.replace('Test')
-    }).catch((error) => {
-      console.log(error)
-      setRegisterError('Email in use. Use a different email')
-    })
+      .then(userCredential => {
+        console.log(userCredential);
```
Comment (YilKanda) [inline AppVuelos/src/screens/register.tsx]: Don't leave debugging code like console.log()

File: AppVuelos/src/screens/register.tsx [hunk 2]
```diff
@@ -39,25 +44,49 @@ function Register(): JSX.Element {
...
+        setTimeout(() => {
+          setComplated(true);
+        }, 1000);
+        setTimeout(() => {
+          setvisible(false);
+          navigation.replace('Login');
+        }, 2000);
+      })
+      .catch(error => {
+        console.log(error);
```
Comment (YilKanda) [inline AppVuelos/src/screens/register.tsx]: Don't leave debugging code like console.log()

File: AppVuelos/src/screens/register.tsx [hunk 3]
```diff
@@ -39,25 +44,49 @@ function Register(): JSX.Element {
   };
 
   const handleButtonClick = async () => {
+    setvisible(true);
     const formData = getFormData(firstName, email, password);
     console.log(formData);
```
Comment (YilKanda) [inline AppVuelos/src/screens/register.tsx]: Don't leave debugging code like console.log()

**Review by YilKanda — DISMISSED** [review]

**Pull request merged**